src/App.tsx

```diff
@@ -1,14 +1,66 @@
-/* eslint-disable max-len */
-import React from 'react';
+import React, { useEffect, useState } from 'react';
 import 'bulma/css/bulma.css';
 import '@fortawesome/fontawesome-free/css/all.css';
 
 import { TodoList } from './components/TodoList';
 import { TodoFilter } from './components/TodoFilter';
+import { getTodos } from './api';
+import { useDispatch } from 'react-redux';
+import { useAppSelector } from './app/hooks';
+import { actions } from './features/todos';
+import { Todo } from './types/Todo';
 import { TodoModal } from './components/TodoModal';
 import { Loader } from './components/Loader';
 
 export const App: React.FC = () => {
+  const dispatch = useDispatch();
+  const { todos } = useAppSelector(state => state.todos);
+  const [showLoader, setShowLoader] = useState(false);
+  const [visibleTodos, setVisibleTodos] = useState<Todo[]>([]);
+  const currentTodo = useAppSelector(state => state.currentTodo);
+  const { query } = useAppSelector(state => state.filter);
+
+  const filter = useAppSelector(state => state.filter);
+
+  useEffect(() => {
+    setShowLoader(true);
+    getTodos()
+      .then(response => {
+        dispatch(actions.setTodos(response));
+      })
+      .finally(() => setShowLoader(false));
+  }, [dispatch]);
+
+  useEffect(() => {
+    if (todos) {
+      setVisibleTodos(todos);
+    }
+  }, [todos]);
+
+  useEffect(() => {
+    if (todos) {
+      let copy: Todo[] = [...todos];
+
+      if (filter.query.length > 0) {
+        copy = copy.filter(elem =>
+          elem.title
+            .toLocaleLowerCase()
+            .includes(filter.query.toLocaleLowerCase()),
+        );
+      }
+
+      if (filter.status === 'completed') {
+        copy = copy.filter(elem => elem.completed === true);
+      }
+
+      if (filter.status === 'active') {
+        copy = copy.filter(elem => elem.completed === false);
+      }
+
+      setVisibleTodos(copy);
+    }
```

Comment:
Consider re-write like that 
```suggestion
    if (!todos) return;
   
      let copy: Todo[] = [...todos];

      if (filter.query.length > 0) {
        copy = copy.filter(elem =>
          elem.title
            .toLocaleLowerCase()
            .includes(filter.query.toLocaleLowerCase()),
        );
      }

      if (filter.status === 'completed') {
        copy = copy.filter(elem => elem.completed === true);
      }

      if (filter.status === 'active') {
        copy = copy.filter(elem => elem.completed === false);
      }

      setVisibleTodos(copy);
```